common/hooks/pre-pkg/04-generate-runtime-deps.sh

```diff
@@ -72,18 +72,17 @@ hook() {
 		    msg_normal "Skipping dependency scan for ${lf}\n"
 		    continue
 	    fi
-        case "$(file -bi "$f")" in
-            application/x-*executable*|application/x-sharedlib*)
-                for nlib in $($OBJDUMP -p "$f"|grep NEEDED|awk '{print $2}'); do
-                    [ -z "$verify_deps" ] && verify_deps="$nlib" && continue
-                    found=0
-                    for j in ${verify_deps}; do
-                        [[ $j == $nlib ]] && found=1 && break
-                    done
-                    [[ $found -eq 0 ]] && verify_deps="$verify_deps $nlib"
+        read -n4 elfmagic < "$f"
+        if [ "$elfmagic" = $'\177ELF' ]; then
+            for nlib in $($OBJDUMP -p "$f"|grep NEEDED|awk '{print $2}'); do
```

Comment:
Maybe replace `grep NEEDED | awk '{print $2}'` with `awk '/NEEDED/{print $2}'` to save some `fork(2)`, `execve(2)`, `pipe(2)` and `dup2`

Reply:
Done and seems to be working!